### PR TITLE
feat(wasm-gen): increase fuzzer stack height limit

### LINKED ISSUE #3532: Revert two stack limiter injections in `wasm-gen` in favor of increasing stack height for fuzz tests
Revert this #3448 if coverage shows better results with larger stack limiter

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Resolves #3532

Advantages:
- faster fuzzing
- better coverage

Disadvantages:
- different runtime behavior depending on `fuzz` feature

### CODE REVIEW

File: pallets/gear/src/schedule.rs [hunk 1]
```diff
@@ -742,7 +748,10 @@ impl<T: Config> Default for Schedule<T> {
 impl Default for Limits {
     fn default() -> Self {
         Self {
+            #[cfg(not(feature = "fuzz"))]
             stack_height: Some(STACK_HEIGHT_LIMIT),
+            #[cfg(feature = "fuzz")]
+            stack_height: Some(FUZZER_STACK_HEIGHT_LIMIT),
```
Comment (techraed) [inline pallets/gear/src/schedule.rs:754]: what about turning it off completely?

**Review by techraed — COMMENTED** [review]
did you run it? It still gives very few `wasm 'unreachable' instruction` errors?

**Review by techraed — APPROVED** [review]

File: sandbox/sandbox/src/embedded_executor.rs [hunk 2]
```diff
@@ -46,7 +46,15 @@ impl<T> Store<T> {
 
 impl<T> SandboxStore<T> for Store<T> {
```
Comment (breathx) [inline sandbox/sandbox/src/embedded_executor.rs:47]: why?

File: pallets/gear/src/schedule.rs [hunk 3]
```diff
@@ -56,8 +56,14 @@ pub const INSTR_BENCHMARK_BATCH_SIZE: u32 = 500;
 // To avoid potential stack overflow problems we have a panic in sandbox in case,
 // execution is ended with stack overflow error. So, process queue execution will be
 // stopped and we will be able to investigate the problem and decrease this constant if needed.
+#[cfg(not(feature = "fuzz"))]
 pub const STACK_HEIGHT_LIMIT: u32 = 36_743;
 
+// For the fuzzer, we take the maximum possible stack limit calculated by the `calc-stack-height`
```
Comment (breathx) [inline pallets/gear/src/schedule.rs:62]: I guess here and above we need doc comments

File: sandbox/sandbox/src/embedded_executor.rs [hunk 4]
```diff
@@ -46,7 +46,15 @@ impl<T> Store<T> {
...
     fn new(state: T) -> Self {
-        let engine = Engine::default();
+        let register_len = mem::size_of::<UntypedValue>();
+
+        let mut config = Config::default();
+        config.set_stack_limits(
+            StackLimits::new(1024 / register_len, (1024 * 1024) / register_len, 16384)
```
Comment (breathx) [inline sandbox/sandbox/src/embedded_executor.rs:53]: mv magic numbers to const and/or variables above

File: pallets/gear/src/schedule.rs (hunk 3 above)
Comment (breathx) [reply pallets/gear/src/schedule.rs:62]: Still actual

**Review by breathx — APPROVED** [review]

File: sandbox/sandbox/src/embedded_executor.rs (hunk 4 above)
Comment (StackOverflowExcept1on) [reply sandbox/sandbox/src/embedded_executor.rs]: in general, these constants should be in our fork, but we have 4 versions of wasmi in Cargo.lock (see #3214)

Comment (StackOverflowExcept1on) [reply sandbox/sandbox/src/embedded_executor.rs]: temp fix is 3f1027548d7f9db011acf9eac856c3f61476bb84

**Pull request merged**